fastify.js

```diff
@@ -585,13 +595,23 @@ function build (options) {
         return
       }
 
-      if (opts.beforeHandler) {
-        if (Array.isArray(opts.beforeHandler)) {
-          opts.beforeHandler.forEach((h, i) => {
-            opts.beforeHandler[i] = h.bind(_fastify)
-          })
+      if (opts.preHandler == null && opts.beforeHandler != null) {
+        beforeHandlerWarning()
+        opts.preHandler = opts.beforeHandler
+      }
+      if (opts.preHandler) {
+        if (Array.isArray(opts.preHandler)) {
+          opts.preHandler.map(hook => hook.bind(_fastify))
```

Comment:
This should be:

```js
opts.preHandler = opts.preHandler.map(hook => hook.bind(_fastify))
```

It looks like there aren't any tests to test that `this` is the fastify instance inside `preHandler/preValidation` hooks that are passed to `.route()` as an array. It would be a good idea to add a test to catch this case.

Reply:
That test wasn't failing because `preHandler` was set as a function. If it were an array of functions it would have failed.